python/fastquant/strategies/base.py

```diff
@@ -289,6 +294,33 @@ def next(self):
                         self.sell(exectype=bt.Order.StopTrail, trailpercent=self.stop_trail, size=final_size)
 
         elif self.sell_signal():
+            if self.allow_short == True:
+
+                # Sell short based on the closing price of the previous day
+                if self.execution_type == "close":
+
+                    if self.transaction_logging:
+                        self.log("SELL CREATE, %.2f" % self.dataclose[0])
+
+                    sell_prop_size = int(self.short_max * 
```

Comment:
I think this shouldn't just be based on cash, but on the total value of the existing portfolio (cash and stock holdings). Also, the sizing here should still be based on `BUY_PROP`, but the basis of the absolute value to sell should be based on the portfolio value.

`self.short_max` should be used as a natural *maximum* to the amount you're allowed to short sell. I'm actually feeling partial towards implementing this as a multiplier on constant number, maybe `init_cash` rather than something moving like your cash balance.